lib/std/heap/uefi_allocators.zig

```diff
@@ -0,0 +1,302 @@
+const std = @import("../std.zig");
+
+const mem = std.mem;
+const uefi = std.os.uefi;
+
+const Allocator = mem.Allocator;
+
+const assert = std.debug.assert;
+
+pub var global_page_allocator = PageAllocator{};
+pub var global_pool_allocator = PoolAllocator{};
+
+/// Allocates memory in pages.
+///
+/// This allocator is backed by `allocatePages` and is therefore only suitable for usage when Boot Services are available.
+pub const PageAllocator = struct {
```

Comment:
The namespacing here is awkward, this ought to be a file-as-struct (`UefiPageAllocator.zig`).

Reply:
Why file as a struct when there's multiple allocators?

Reply:
Because `std.heap.UefiPageAllocator` is much nicer than `std.heap.uefi_allocators.PageAllocator` (which violates https://ziglang.org/documentation/master/#Avoid-Redundant-Names-in-Fully-Qualified-Namespaces). Why do you want both allocators in the same file?

Reply:
> Because `std.heap.UefiPageAllocator` is much nicer than `std.heap.uefi_allocators.PageAllocator`

Wouldn't `std.heap.uefi.Page/Pool` be nicer?

> Why do you want both allocators in the same file?

Just so there isn't 3 more files that could fit into 1.

Reply:
> Wouldn't std.heap.uefi.Page/Pool be nicer?

Not really, a "page" and a "page allocator" are different things. Avoiding redundancy does not mean to leave out important bits of information until the name has a different meaning.

> Just so there isn't 3 more files that could fit into 1.

That's not a good reason.

Reply:
Ok, I've managed to find the time and split the allocators into their own files.